tests/unit/test_parse_listener.py

```diff
@@ -0,0 +1,23 @@
+import antlr4.tree.Tree
+from antlr4 import InputStream
+
+from pytsql.grammar import TSqlParserListener, parse
+
+
+class TestTSqlParserListener(TSqlParserListener.ParseTreeListener):
+    def __init__(self):
+        self.called = False
+
+    def enterTsql_file(self, ctx):
+        self.called = True
+
+
+def test_parser_listener_called_at_enter_tsql_file():
+    seeed = "SELECT * FROM tbl"
+
+    tree = parse(InputStream(data=seeed), "tsql_file")
+    walker = antlr4.ParseTreeWalker()
+    test_listener = TestTSqlParserListener()
```

Comment:
Not necessary to change, just FYI, I think you can also use a [MagicMock](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.MagicMock) object because it's the standard for mocking functionality, as it has many convenience methods (`called` included)
```python
from unittest.mock import MagicMock
...
...
    test_listener = MagicMock()
```